.gitlab/source_test/golang_deps_generate.yml

```diff
@@ -0,0 +1,22 @@
+---
+# golang_deps_generate stage
+# Contains the step to build the golang dependency chain and preserve it
+golang_deps_generate:
+  rules:
+    !reference [.on_main_or_release_branch]
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  needs: ["linux_x64_go_deps"]
+  variables:
+    ARCH: "x64"
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - python3 -m pip install --upgrade --ignore-installed pip setuptools
+    - python3 -m pip install -r requirements.txt
+  script:
+    - inv agent.build-dep-tree
+    - tar -czf dependency_tree.tgz dependency_tree.txt
```

Comment:
Do we really need to `tar` the output? We don't usually do this unless we have several files that we reuse later in the pipeline (not the case here IIUC)

Reply:
The resulting file is a plain text >100MB file. It seemed wasteful not to compress the artifact as it results in a 3-4MB artifact instead.

Reply:
Fair enough :)